app/controllers/favorites_controller.rb

```diff
@@ -0,0 +1,22 @@
+class FavoritesController < ApplicationController
+  before_filter :authenticate_user!
+  before_action :set_post
+
+  def create
+    favorite = Favorite.new(user_id: current_user.id, post_id: @post.id)
```

Comment:
I just took the post object out. This controller only ever uses the id of post, so I don't think there's a need to even look up post in the first place. Thanks for pointing this out.